src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,40 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Class representing Deadlines
+ */
+public class Deadline extends Task {
+    protected String by;
+
+    protected LocalDate byDate;
+
+    /**
+     * Creates an instance of Deadline
+     * @param name Name of the deadline
+     * @param by Due date of the deadline
+     */
+    public Deadline(String name, String by) {
+        super(name);
+        this.by = by;
+
+        // Check if they are in dateTime format
+        try {
+            byDate = LocalDate.parse(by);
+            this.by = byDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        } catch (DateTimeParseException e) {}
+    }
```

Comment:
Try and avoid having too much business logic in class constructors. The input should already be parsed and processed by the Parser.java class instead of letting Deadline.java handle it